src/main/app/src/app/informatie-objecten/informatie-object-add/informatie-object-add.component.ts

```diff
@@ -333,23 +333,29 @@ export class InformatieObjectAddComponent implements AfterViewInit, OnDestroy {
       Object.keys(formGroup.controls).forEach((key) => {
         const control = formGroup.controls[key];
         const value = control.value;
-        if (value instanceof moment) {
-          infoObject[key] = value; // conversie niet nodig, ISO-8601 in UTC gaat goed met java ZonedDateTime.parse
-        } else if (key === "informatieobjectTypeUUID") {
-          infoObject[key] = value.uuid;
-        } else if (key === "taal") {
-          infoObject[key] = value.code;
-        } else if (key === "status") {
-          infoObject[key] = InformatieobjectStatus[value.value.toUpperCase()];
-        } else if (key === "vertrouwelijkheidaanduiding") {
-          infoObject[key] = value.value;
-        } else if (key === "bestand") {
-          infoObject["bestandsomvang"] = value.size;
-          infoObject["bestandsnaam"] = value.name;
-          infoObject["bestand"] = value;
-          infoObject["formaat"] = value.type;
-        } else {
-          infoObject[key] = value;
+
+        switch (key) {
+          case "informatieobjectTypeUUID":
```

Comment:
Keys like `informatieobjectTypeUUID` are also defined in `zac-openapi-types.d.ts` (which is automatically generated from the backend API). Is it not possible to use those here I wonder?

Reply:
Would be great if possible! I talked with Bas about this and just mentioned this improvement to Karin. We should use some generated type safety indeed. I am curious in what is exactly at hand and we can use already.

Reply:
We started generating this code to gain more type safety indeed (Rick initially set this up). As far as I know we use it in several places already.

Reply:
But not open-api'd in this task for sure! @edgarvonk

Reply:
I'm not sure exactly how those OpenAPI types are used in the frontend code. Would be good to delve into that. :-)

Reply:
In docs/development/endToEndTypeSafety.md I read:

_**Good TypeScript code is written with as little TypeScript code as possible.**_

I need to take a moment to digest 😉